sematic/api/endpoints/meta.py

```diff
@@ -51,3 +57,59 @@ def env_endpoint(user: Optional[User]) -> flask.Response:
             continue
 
     return flask.jsonify(dict(env=env))
+
+
+@sematic_api.route("/api/v1/meta/health", methods=["GET"])
+def get_server_health() -> flask.Response:
+    """Get information about the server health.
+
+    Returns
+    -------
+    A flask response with a json payload. The payload has two fields:
+        api: status info about basic access to the API
+        db: status info about access to the DB
+    Status info for each of these fields is a dict of the form:
+    {
+        "healthy": <bool>,
+        "message": <str message about status>,
+    }
+    If any of the "healthy" fields is `False`, will return a 503 status code.
+    """
+    # healthy by nature of the fact that this API is working.
+    api_status = dict(
+        healthy=True,
+        message="API is healthy",
+    )
+
+    db_healthy = True
+    db_message = "Basic database access verified"
+    with db().get_session() as session:
+        try:
+            statement = sqlalchemy.text("SELECT id FROM runs LIMIT 1;")
+            if len(list(session.execute(statement))) > 0:
+                db_message = "Verified at least one run is present in the DB."
+
+        except Exception as e:
+            logger.fatal(
+                "Sematic is unable to access the database and "
+                "therefore will be unable to operate properly: %s",
+                e,
+            )
+            db_healthy = False
+            db_message = f"Problems accessing the database. Error: {e}"
+
+    db_status = dict(
+        healthy=db_healthy,
+        message=db_message,
+    )
+    payload = dict(
+        api=api_status,
+        db=db_status,
+    )
+
+    status = HTTPStatus.OK if db_healthy else HTTPStatus.SERVICE_UNAVAILABLE
```

Comment:
It was actually determined to be a desired behavior that Sematic should go down if we lose DB access. Sematic will fail to do anything anyway, but it will just show up as a bunch of 500 errors on other unrelated endpoints. This makes the signal clear that Sematic went down because it couldn't reach the DB.

Reply:
yea i agree with @tscurtu on this one.  we need to have two separate health endpoints.  one is for the sematic server itself being live and healthy, which is what we had already, and that endpoint should be used by k8s to keep pods running and linked to the service.

the second endpoint should indicate health of the sematic server's dependencies, and this endpoint should be queried by the front-end code to probe for status.  this endpoint is what you've implemented here.  note that this endpoint explicitly _cannot_ be the same one that k8s uses to determine app health, because if it starts failing, front-end requests for health status will no longer get routed to any k8s pods since they will not be associated with the service.

i don't think we want sematic to go down if we lose db access.  i'm not sure where that determination came from.

Reply:
To this point, please either rename the endpoint to not suggest this determines the health of the Server itself, or split it into 2 endpoints: the health of the Server itself, and the health of dependencies.

This way, the Server instance won't be terminated by monitoring solutions if only dependencies are down.